osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/GetPeriodicMeterReadsCommandExecutorTest.java

```diff
@@ -110,43 +163,74 @@ public void testHappy() throws Exception {
         final GetResult result3 = mock(GetResult.class);
         final GetResult result4 = mock(GetResult.class);
         final GetResult result5 = mock(GetResult.class);
+
+        final GetResult getResult = mock(GetResult.class);
+
         when(this.dlmsHelper.getAndCheck(this.connectionManager, device, expectedDescription, attributeAddress))
                 .thenReturn(asList(result0, result1, result2, result3, result4, result5));
 
         when(this.dlmsHelper.readDataObject(result0, PERIODIC_E_METER_READS)).thenReturn(resultData);
 
+        when(this.dlmsHelper.getAndCheck(eq(this.connectionManager), eq(this.device), eq(expectedDescription),
+                eq(attributeAddressForProfile.getAttributeAddress()))).thenReturn(Collections.singletonList(getResult));
+        when(this.dlmsHelper.getAndCheck(this.connectionManager, this.device, expectedDescription,
+                attributeAddress)).thenReturn(Collections.singletonList(getResult));
+
+        when(this.dlmsHelper.readDataObject(eq(getResult), any(String.class))).thenReturn(resultData);
+
         final CosemDateTimeDto cosemDateTime = mock(CosemDateTimeDto.class);
         final String expectedDateTimeDescription = String.format("Clock from %s buffer", periodType);
         when(this.dlmsHelper.readDateTime(data0, expectedDateTimeDescription)).thenReturn(cosemDateTime);
+
         final DateTime bufferedDateTime = DateTime.now();
         when(cosemDateTime.asDateTime()).thenReturn(bufferedDateTime);
 
-        // TODO: mock other calls
-
-        final PeriodicMeterReadsRequestDto request = new PeriodicMeterReadsRequestDto(periodType, new Date(from),
-                new Date(to));
+        final DlmsMeterValueDto meterValue1 = mock(DlmsMeterValueDto.class);
+        final DlmsMeterValueDto meterValue2 = mock(DlmsMeterValueDto.class);
+        when(this.dlmsHelper.getScaledMeterValue(data1, null, "positiveActiveEnergyTariff1")).thenReturn(meterValue1);
+        when(this.dlmsHelper.getScaledMeterValue(data4, null, "positiveActiveEnergyTariff2")).thenReturn(meterValue2);
```

Comment:
Hier kan je alle waardes mocken:
        when(this.dlmsHelper.getScaledMeterValue(data1, null, "positiveActiveEnergyTariff1")).thenReturn(meterValue1);
        when(this.dlmsHelper.getScaledMeterValue(data2, null, "positiveActiveEnergyTariff2")).thenReturn(meterValue2);
        when(this.dlmsHelper.getScaledMeterValue(data3, null, "negativeActiveEnergyTariff1")).thenReturn(meterValue3);
        when(this.dlmsHelper.getScaledMeterValue(data4, null, "negativeActiveEnergyTariff2")).thenReturn(meterValue4);

Reply:
done

Reply:
This violates the 'Law of Demeter', see the famous post: https://tuhrig.de/everytime-a-mock-returns-a-mock-a-fairy-dies/